[NFC] Clean Up Some Unneeded TypeCheckerOptions

CodaFi opened the pull request:
While refactoring in 48805b1, I accidentally added the computation of this bit before CompilerInstance::setupInputs is called. This means that the compiler currently does not have any knowledge of any primary input buffers, and thus the check for whole module mode is trivially true. As a consequence, this bit has been true ever since.

Since we seem to have got on just fine without computing this correctly, just inline that truthiness everywhere.

Whoops

Also, while I'm here, we don't need `InImmediateMode`.

CodaFi committed:
[NFC] Remove DelayWholeModuleChecking

While refactoring in 48805b1, I accidentally added the computation of this bit before CompilerInstance::setupInputs is called. This means that the compiler currently does not have any knowledge of any primary input buffers, and thus the check for whole module mode is trivially true. As a consequence, this bit has been true ever since.

Since we seem to have got on just fine without computing this correctly, just inline that truthiness everywhere.

Whoops

CodaFi committed:
[Gardening] Remove InImmediateMode

The last reader of this bit was the legacy integrated REPL.

CodaFi commented:
@swift-ci test

CodaFi commented on lib/Frontend/Frontend.cpp:
```diff
@@ -892,11 +883,9 @@ void CompilerInstance::forEachFileToTypeCheck(
-      performWholeModuleTypeChecking(SF);
-    });
-  }
+  forEachSourceFileIn(MainModule, [&](SourceFile &SF) {
```
I *think* this always should have been `forEachFileToTypeCheck`. Gonna give that a shot locally. Would certainly be a win in incremental mode.

davidungar reviewed (approved):
Looks pretty good. Hope it works!

CodaFi committed:
[NFC] Remove forEachSourceFileIn

We actually wanted to use forEachFileToTypeCheck, which will restrict
the source files we analyze to just the given primaries in incremental
mode, and the whole module's source files otherwise.

CodaFi commented:
@swift-ci test

CodaFi commented:
@swift-ci test

CodaFi commented:
⛵
